src/Domain/Block.cpp

```diff
@@ -11,7 +11,6 @@
 #include <typeinfo>
```

Comment:
I don't really know how to review this commit. The commit message doesn't help me understand why the domain creators are no longer having the boundary conditions. Where are they going? This commit just does a lot and would be a lot easier to review if it's possible to split into smaller logical pieces...

Reply:
The domain creators still have the boundary conditions! They have just moved from `create_domain()` to the new `external_boundary_conditions()` function, and then they are stored in `domain::Tags::ExternalBoundaryConditions`.

This commit is definitely annoying to review (and the refactoring has been pretty annoying too 😅). I can split it up into a few logical commits and squash them together later if that helps?

Reply:
Ah okay! I see the pattern now :) Can't (yet) promise I will understand the test changes, but I can now understand the `src` changes, so leave it as-is for now :)